src/pages/workspaces/workspace/useWorkspace.ts

```diff
@@ -0,0 +1,191 @@
+      await loadGoogleBucketLocation(workspace)
+    } else if (!!workspace && isAzureWorkspace(workspace)) {
+      // console.log('Azure, skipping storage initialization check')
+      await loadAzureStorageDetails(workspace)
+    }
+  }
+
+  const checkGooglePermissions = async workspace => {
+    try {
+      await Ajax(signal).Workspaces.workspace(namespace, name).checkBucketReadAccess()
+      // console.log('got success status!!!!!!')
+      updateWorkspaceInStore(workspace, true)
+      loadGoogleBucketLocation(workspace)
+    } catch (error) {
+      // @ts-ignore
+      const errorText = await error.text()
+      if (responseContainsRequesterPaysError(errorText)) {
+        updateWorkspaceInStore(workspace, true)
+      } else {
+        updateWorkspaceInStore(workspace, false)
+        console.log('Google permissions are still syncing') // eslint-disable-line no-console
+        checkInitializationTimeout.current = setTimeout(() => checkWorkspaceInitialization(workspace), GOOGLE_PERMISSIONS_RECHECK_RATE)
+      }
+    }
+  }
+
+  // Note that withErrorIgnoring is used because checkBucketLocation will error for requester pays workspaces.
+  const loadGoogleBucketLocation = withErrorIgnoring(async workspace => {
+    const storageDetails = await Ajax(signal).Workspaces.workspace(namespace, name).checkBucketLocation(workspace.workspace.googleProject, workspace.workspace.bucketName)
+    // console.log(`setting bucketLocation ${storageDetails.location}`)
+    setGoogleStorage(storageDetails)
+  })
+
+  const storeAzureStorageDetails = azureStorageDetails => {
+    const { location, sas } = azureStorageDetails
+    const sasUrl = sas.url
+    setAzureStorage({ storageContainerUrl: _.head(_.split('?', sasUrl)), location, sasUrl })
+  }
+
+  const checkAzureStorageExists = async workspace => {
+    try {
+      storeAzureStorageDetails(await Ajax(signal).AzureStorage.details(workspace.workspace.workspaceId))
+      // console.log('got success status!!!!!!')
+      updateWorkspaceInStore(workspace, true)
+    } catch (error) {
+      updateWorkspaceInStore(workspace, false)
+      // We expect to get a transient error while the workspace is cloning. We will improve
+      // the handling of this with WOR-534 so that we correctly differentiate between the
+      // expected transient error and a workspace that is truly missing a storage container.
+      console.log(`Error thrown by AzureStorage.details: ${error}`) // eslint-disable-line no-console
+      checkInitializationTimeout.current = setTimeout(() => checkWorkspaceInitialization(workspace), AZURE_BUCKET_RECHECK_RATE)
+    }
+  }
+
+  const loadAzureStorageDetails = withErrorReporting('Error loading storage information', async workspace => {
+    storeAzureStorageDetails(await Ajax(signal).AzureStorage.details(workspace.workspace.workspaceId))
+  })
+
+  const refreshWorkspace = _.flow(withErrorReporting('Error loading workspace'), Utils.withBusyState(setLoadingWorkspace))(async () => {
```

Comment:
Much of this code came from [WorkspaceContainer](https://github.com/DataBiosphere/terra-ui/pull/3765/files#diff-1f45f11cb74b4b6ee8c1c20f21df6fedeebdf53163ac7589148d5f43867534d6L348).